imap_processing/tests/conftest.py

```diff
@@ -155,3 +159,79 @@ def _unset_metakernel_path(monkeypatch):
     """Temporarily unsets the SPICE_METAKERNEL environment variable"""
     if os.getenv("SPICE_METAKERNEL", None) is not None:
         monkeypatch.delenv("SPICE_METAKERNEL")
+
+
+@pytest.fixture()
+def generate_spin_data():
+    def make_data(start_met: int, end_met: Optional[int] = None) -> pd.DataFrame:
+        """
+        Generate a spin table CSV covering one or more days.
+        Spin table contains the following fields:
+            (
+            spin_number,
+            spin_start_sec,
+            spin_start_subsec,
+            spin_period_sec,
+            spin_period_valid,
+            spin_phas_valid,
+            spin_period_source,
+            thruster_firing
+            )
+        This function creates spin data using start MET and end MET time.
+        Each spin start data uses the nominal 15 second spin period. The spins that
+        occur from 00:00(Mid-night) to 00:10 UTC are marked with flags for
+        thruster firing, invalid spin period, and invalid spin phase.
+        Parameters
+        ----------
+        start_met : int
+            Provides the start time in Mission Elapsed Time (MET).
+        end_met : int
+            Provides the end time in MET. If not provided, default to one day
+            from start time.
+        Returns
+        -------
+        spin_df : pd.DataFrame
+            Spin data.
+        """
+        if end_met is None:
+            # end_time is one day after start_time
+            end_met = start_met + 86400
+
+        # Create spin start second data of 15 seconds increment
+        spin_start_sec = np.arange(start_met, end_met, 15)
+
+        spin_dict = {
+            "spin_number": np.arange(spin_start_sec.size, dtype=np.uint32),
+            "spin_start_sec": spin_start_sec,
+            "spin_start_subsec": np.full(spin_start_sec.size, 0, dtype=np.uint32),
+            "spin_period_sec": np.full(spin_start_sec.size, 15.0, dtype=np.float32),
+            "spin_period_valid": np.ones(spin_start_sec.size, dtype=np.uint8),
+            "spin_phas_valid": np.ones(spin_start_sec.size, dtype=np.uint8),
+            "spin_period_source": np.zeros(spin_start_sec.size, dtype=np.uint8),
+            "thruster_firing": np.zeros(spin_start_sec.size, dtype=np.uint8),
+        }
+
+        # Convert spin_start_sec to datetime to set repointing times flags
+        spin_start_dates = met_to_j2000ns(spin_start_sec)
+        spin_start_dates = cdflib.cdfepoch.to_datetime(spin_start_dates)
+
+        # Convert DatetimeIndex to Series for using .dt accessor
+        spin_start_dates_series = pd.Series(spin_start_dates)
+
+        # Find index of all timestamps that fall within 10 minutes after midnight
+        repointing_times = spin_start_dates_series[
+            (spin_start_dates_series.dt.time >= pd.Timestamp("00:00:00").time())
+            & (spin_start_dates_series.dt.time < pd.Timestamp("00:10:00").time())
+        ]
```

Comment:
Your "00" test will always be true, so you can either drop that one, or say +/- 10 minutes so you get before and after midnight.
```suggestion
        # Find index of all timestamps that fall within 10 minutes of midnight
        repointing_times = spin_start_dates_series[
            (spin_start_dates_series.dt.time > pd.Timestamp("23:50:00").time())
            & (spin_start_dates_series.dt.time < pd.Timestamp("00:10:00").time())
        ]
```

Reply:
I am assuming nominal repointing time is 10 minutes. With this, it will become 20 minutes. Is that right? Did you want this to five minutes before and after instead.